docs/querying/sql.md

```diff
@@ -561,6 +561,8 @@ The [DataSketches extension](../development/extensions-core/datasketches-extensi
 |`COALESCE(value1, value2, ...)`|Returns the first value that is neither NULL nor empty string.|
 |`NVL(expr,expr-for-null)`|Returns 'expr-for-null' if 'expr' is null (or empty string for string type).|
 |`BLOOM_FILTER_TEST(<expr>, <serialized-filter>)`|Returns true if the value is contained in a Base64-serialized bloom filter. See the [Bloom filter extension](../development/extensions-core/bloom-filter.html) documentation for additional details.|
+|`CONTAINS_STR(<expr>, str)`|Returns true if the `str` is a substring of `expr`.|
+|`ICONTAINS_STR(<expr>, str)`|Returns true if the `str` is a substring of `expr`. The match is case-insensitive.|
```

Comment:
The PR description says the functions are called `contains` and `icontains`, but it looks like this patch uses the names with `_str` appended. Was there any reason you chose these names instead of others?

I couldn't find an equivalent query in postgresql, and when I googled for a contains operator: Oracle and SQL server appear to have one - https://docs.oracle.com/cd/B28359_01/text.111/b28303/query.htm#g1016054 and https://docs.microsoft.com/en-us/sql/t-sql/queries/contains-transact-sql?view=sql-server-2017. 

Also, was there any reasoning behind having 2 functions instead of 1 with a parameter to ignore the case? I don't think I have a strong preference either way yet, just curious

Reply:
These should be in the "String functions" section.

I don't have an opinion on the names right now, but to form an opinion I like to do the following.

1. Check Calcite's SqlStdOperatorTable, which contains standard operators. If there's something matching there, we should use that.
2. If there isn't a standard operator, survey some other databases. I like to check the golden oldies: MySQL, PostgreSQL, Oracle, SQL Server, as well as some of the newer ones like Presto, Snowflake, BigQuery. If a few of them seem to agree on a name and behavior it's good to go with that.
3. If there doesn't seem to be any agreement, or if you don't want to implement what they seem to agree on for some reason, then make up a new operator.

Reply:
OK thinking about this some more, I think I like the idea of one function name more since it maps to the `SearchQuerySpec`. Then, over time, this function can become smarter and support the different types of search query specs, so if a user passes in a list of values, it knows to use the `FragmentSearchQuerySpec` and if they pass in some other flag, it knows to use the `RegexSearchQuerySpec`

Reply:
These are really good points and it's my bad to not put my reasonings in the description earlier.  The description has more details but in short, 
 - `contains` is a standard operator in calcite, built with non-string input types. 
 - The function names are different for case insensitive flavour, to not overload the function signature and follow standard MySQL convention (`like`/`ilike`)

Reply:
Did some collection of those other databases I mentioned seem to agree on a way to do this?

If not, I think your reasoning makes sense. The only thing I'd consider changing, in that case, is spelling out STRING rather than STR. We have a few other functions that have STRING in the name and none that have STR: https://druid.apache.org/docs/latest/querying/sql

On the two-functions vs extra-parameter thing: IMO it feels more SQL-ish to have two functions. It doesn't map as directly onto the native construct, but I think that's okay. The native constructs were designed for a JSON API, where it's easy and natural to add a bunch of named parameters. That isn't the case with SQL, where parameters are positional, and so adding a lot of them doesn't work as well.

Reply:
mysql and postgres - They have substring functions that return the starting index of a search string. I did not find any function similar to `contains`. Oracle has a similar function which is called `instr` 
Snowflake has a `contains` function similar to ours though there is no case insensitive flavor. 
Presto does not seem to have a `contains` method but does have the methods to find the starting index of another string.
BigQuery has `REGEXP_CONTAINS` which takes a regular expression as an input. 

`CONTAINS` would have been an ideal choice for the name but alas, that conflicts with the existing calcite operator.  I can certainly rename them to `CONTAINS_STRING`